R/package.R

```diff
@@ -178,13 +178,33 @@ pkgdown_config_path <- function(path) {
   )
 }
 
-read_meta <- function(path) {
+read_meta <- function(path, call = caller_env()) {
   path <- pkgdown_config_path(path)
 
   if (is.null(path)) {
     yaml <- list()
   } else {
-    yaml <- yaml::yaml.load_file(path) %||% list()
+    yaml <- withCallingHandlers(
+      yaml::yaml.load_file(path) %||% list(),
+      error = function(e) {
+        # Tweak the original message to put the location of the error at the end
+        # based on yaml 2.3.8 error message structure
+        # (<<path>>) Parser error: <<parsing error>>
+        yaml_err <- conditionMessage(e)
+        # extract parsing error from original error (i.e. remove the path)
+        parsing_error <- sub("[^\\)]+\\)\\s?", "", yaml_err)
+        # Extract path from original error
+        path_yaml <- regmatches(yaml_err, m = regexpr("\\(([^\\)]+)\\)", yaml_err))
+        path_yaml <- gsub("\\(([^\\)]+)\\)", "\\1", path_yaml)
+        # Rethrow cli-styled error!
+        cli::cli_abort(c(
+          "x" = "Could not parse the config file.",
+          "!" = parsing_error,
+          "i" = "Edit {.path {path_yaml}} to fix the problem."
+          ),
+          call = call
+        )
```

Comment:
I'd suggest making this a separate function, so it's a little easier to test.

Could you please also consider the case where the regex fails (i.e. if the underlying error format happens to change) and then just fall back to the unwrapped error?

Reply:
I found even a better way! passing `error.label = NULL` to `yaml.load_file()` suppresses the path part. So we can just append it. No more regexps!